pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -100,7 +100,8 @@ func (r *ReconcileMachineHealthCheck) Reconcile(request reconcile.Request) (reco
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
+		glog.Warningf("Requeuing the key. Error reading the object: %v", err)
```

Comment:
why is this needed? by default any requests will requeue when returning error != nil